src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1161,13 +1161,18 @@ internal bool FilterAndAppendAndFreeDiagnostics(DiagnosticBag accumulator, ref D
         /// Filter out warnings based on the compiler options (/nowarn, /warn and /warnaserror) and the pragma warning directives.
         /// </summary>
         /// <returns>True when there is no error.</returns>
-        internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming)
+        internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming, PooledHashSet<int> exclude)
```

Comment:
>PooledHashSet<int> [](start = 110, length = 18)

Consider using ```HashSet<int>``` as the type so that consumers of the API were not forced to utilize a pull when it is not necessary.